kotlintest-assertions/src/main/kotlin/io/kotlintest/AnyThrowableHandling.kt

```diff
@@ -15,6 +15,23 @@ package io.kotlintest
  */
 inline fun shouldThrowAnyUnit(block: () -> Unit) = shouldThrowAny(block)
 
+/**
+ * Verifies that a block of code does NOT throw any [Throwable]
+ *
+ * Use function to wrap a block of code that you want to make sure doesn't throw any [Throwable],
+ * where using [shouldNotThrowAny] can't be used for any reason, such as an assignment of a variable (assignments are
+ * statements, therefore has no return value).
+ *
+ * Note that executing this code is not very different than executing [block] by itself, as any uncaught exception will
```

Comment:
* Note that executing this code is no different to executing the [block] by itself, as any uncaught exception will...

Reply:
It's somewhat different, as it will wrap exceptions in an AssertionError, but I'll change that